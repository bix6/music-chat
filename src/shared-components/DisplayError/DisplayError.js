import React from 'react';

const errorStyles = {
    color: 'red',
    fontSize: '1.8rem',
    margin: '1rem 0',
}

export default function DisplayError(props) {
    if (props.message) {
        return (
            <div style={ errorStyles }>{ props.message }</div>
        );
    }

    return <></>;
}