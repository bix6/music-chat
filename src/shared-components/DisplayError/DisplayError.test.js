import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import DisplayError from "./DisplayError";

describe("DisplayError Component", () => {
  it("Smoke Test: Renders Default", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <DisplayError />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot Test: Default", () => {
    const wrapper = shallow(<DisplayError />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
