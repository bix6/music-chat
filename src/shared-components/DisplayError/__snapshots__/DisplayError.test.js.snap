// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayError Component Snapshot Test: Default 1`] = `<Fragment />`;
